### PR TITLE
[Feature] [55608] Create UI for Search Screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 55608: Create UI for Search Screen](https://edu-redmine.sun-asterisk.vn/issues/55608)
## WHAT
Tạo UI:
- Danh sách Category: CollectionView
- Danh sách phim với mỗi item là 1 info của phim: CollectionView
- SearchBar
## Evidence
Simulator Screen Shot - iPhone 14 Pro - 2022-11-14 at 08 34 43
(Screenshot or Video)

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: MovieDB/ViewController/CategoryVC/Cell/SearchCollectionViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//
+//  Created by le.n.t.trung on 11/11/2022.
+//
+
+import UIKit
+
+final class SearchCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet private weak var categoryTitleLabel: UILabel!
+    @IBOutlet private weak var categoryUIView: UIView!
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/Cell/SearchCollectionViewCell.swift]:
```suggestion
    @IBOutlet private weak var categoryView: UIView!
```

File: MovieDB/ViewController/CategoryVC/SearchViewController.swift [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+//  MovieDB
+//
+//  Created by le.n.t.trung on 03/11/2022.
+//
+
+import UIKit
+
+final class SearchViewController: UIViewController {
+    
+    @IBOutlet private weak var appBarUIView: AppBarUIView!
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/SearchViewController.swift]:
```suggestion
    @IBOutlet private weak var appBarView: AppBarUIView!
```

File: MovieDB/ViewController/CategoryVC/SearchViewController.swift [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var categoryCollectionView: UICollectionView!
+    @IBOutlet private weak var searchUIView: UIView!
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/SearchViewController.swift]:
```suggestion
    @IBOutlet private weak var searchView: UIView!
```

File: MovieDB/ViewController/CategoryVC/SearchViewController.swift [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        2
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if section == 0 {
+            return genres.count
+        }
+        return films.count
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/SearchViewController.swift]:
```suggestion
        return section == 0 ? genres.count : films.count
```

File: MovieDB/ViewController/CategoryVC/SearchViewController.swift [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                withReuseIdentifier: "SearchCollectionViewCell",
+                for: indexPath) as? SearchCollectionViewCell else {
+                return UICollectionViewCell()
+            }
+            let selected = genres[indexPath.row].id == selectedGenre?.id ?? -1
+            cell.bindData(genre: genres[indexPath.row], isSelected: selected)
+            return cell
+        }
+        guard let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: "HorizontalCustomCollectionViewCell",
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/SearchViewController.swift]: qua mấy cái `cell` define identifier cho nó e, qua đây gọi ra

**Review by tunghs-1253 — APPROVED** [review]

File: MovieDB/ViewController/CategoryVC/SearchViewController.swift [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        searchView.clipsToBounds = true
+        searchView.layer.cornerRadius = 10
+    }
+    
+    private func registerInit() {
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.backgroundColor = .black
+        
+        collectionView.register(UINib(nibName: "HorizontalCustomCollectionViewCell",
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/SearchViewController.swift]: sửa đây lun e

File: MovieDB/ViewController/CategoryVC/SearchViewController.swift [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+                                      bundle: nil),
+                                forCellWithReuseIdentifier: HorizontalCustomCollectionViewCell.identifier)
+        collectionView.register(UINib(nibName: "SearchCollectionViewCell",
```
Comment (ghost) [inline MovieDB/ViewController/CategoryVC/SearchViewController.swift]: đây nữa

**Review by ghost — APPROVED** [review]

**Pull request merged**